storage/local/storage_test.go

```diff
@@ -204,6 +204,74 @@ func TestRetentionCutoff(t *testing.T) {
 	}
 }
 
+func TestDropMetrics(t *testing.T) {
+	now := clientmodel.Now()
+	insertStart := now.Add(-2 * time.Hour)
+
+	s, closer := NewTestStorage(t, 1)
+	defer closer.Close()
+
+	m1 := clientmodel.Metric{clientmodel.MetricNameLabel: "test", "n1": "v1"}
+	m2 := clientmodel.Metric{clientmodel.MetricNameLabel: "test", "n1": "v2"}
+
+	N := 120000
+
+	samples := make(clientmodel.Samples, N)
+	for j, m := range []clientmodel.Metric{m1, m2} {
+		for i := range samples {
+			smpl := &clientmodel.Sample{
+				Metric:    m,
+				Timestamp: insertStart.Add(time.Duration(i) * time.Millisecond), // 1 minute intervals.
+				Value:     clientmodel.SampleValue(j),
+			}
+			s.Append(smpl)
+		}
+	}
+	s.WaitForIndexing()
+
+	fps := s.FingerprintsForLabelMatchers(metric.LabelMatchers{{
+		Type:  metric.Equal,
+		Name:  clientmodel.MetricNameLabel,
+		Value: "test",
+	}})
+	if len(fps) != 2 {
+		t.Fatalf("unexpected number of fingerprints: %d", len(fps))
+	}
+
+	it := s.NewIterator(fps[0])
+	if vals := it.RangeValues(metric.Interval{insertStart, now}); len(vals) != N {
+		t.Fatalf("unexpected number of samples: %d", len(vals))
+	}
+	it = s.NewIterator(fps[1])
+	if vals := it.RangeValues(metric.Interval{insertStart, now}); len(vals) != N {
+		t.Fatalf("unexpected number of samples: %d", len(vals))
+	}
+
+	s.DropMetricsForFingerprints(fps[0])
+	s.WaitForIndexing()
+
```

Comment:
Maybe also re-check `FingerprintsForLabelMatchers()` to verify that the index deletion worked here and after the second drop.